lib/mix/tasks/group_uploader.ex

```diff
@@ -0,0 +1,80 @@
+defmodule Mix.Tasks.GroupUploader do
+  use Mix.Task
+
+  alias Cadet.{Accounts, Course}
+
+  @moduledoc """
+  Mix Task to upload the tutorial groups and it's
+  participants
+  """
+
+  def run([groups_arg | [avengers_arg | _]]) do
+    Mix.Task.run("app.start")
+    # Removing unneccesary headers
+    groups =
+      groups_arg
+      |> Xlsxir.multi_extract()
+      |> Keyword.get(:ok)
+      |> Xlsxir.get_list()
+      |> Enum.drop(3)
+
+    avengers =
+      avengers_arg
+      |> Xlsxir.multi_extract()
+      |> Keyword.get(:ok)
+      |> Xlsxir.get_list()
+      |> Enum.drop(1)
+
+    upload_to_database(groups, avengers, nil)
+  end
+
+  def run(_), do: {:error, "Invalid arguments"}
+
+  defp upload_to_database([row | rows], avengers, current_group) do
+    cond do
```

Comment:
Please refactor. DO NOT use `cond` unless as a last resort. The whole function is still rather unreadable. Be explicit about what you are trying to achieve, and then comment whenever you feel it's not clear enough.